### DISCUSSION TITLE
Countdown to 10 years of c:geo opensource

### DISCUSSION THREAD

**Discussion by Lineflyer** [cgeo/cgeo#10359] [Show and tell]
I have just realized its almost 10 years for me contributing here, so I crawled the commit and release history and have seen:
- The first tagged release was for 26th, August 2011 with  `market_20110826`  https://github.com/cgeo/cgeo/releases/tag/market_20110826
- The first commit to this repository was done on 11th, July 2011 according to 
```
$ git log --reverse
commit 3a311f2a45a79a19ebc4e2b66f5e92a2f390c04d
Author: mucek4 <tomaz@gorenc.org>
Date:   Mon Jul 11 08:12:48 2011 +0200

    First commit
```
What date would be the birthday of c:geo opensource?

**Comment by Lineflyer** [comment]
BTW:
I took a fight with Github and finally filtered out, what could be my first activity here. It was on 31st, July 2011:
https://github.com/cgeo/cgeo/issues/72

**Comment by SammysHP** [comment]
My first email contact with carnero and mucek4 about c:geo opensource was 2011-07-01.

**Comment by Lineflyer** [comment]
The old one: https://play.google.com/store/apps/details?id=carnero.cgeo

**Comment by rsudev** [comment]
I started with the first commit on 31. May 2011 with developing a mapsforge-based offline map implementation...

**Reply by rsudev** [reply]
The open source story of c:geo started on github on 22 Jul 2010 with the first published commit.

**Reply by Lineflyer** [reply]
Was there a different repo in the beginning?

**Reply by SammysHP** [reply]
Yes, carnero released the source some time before, but without proper license. Several people including me started to work on it. Many of them later became the first "open source team".

My first direct communication with carnero was 2011-01-02 about cache detail and direction encryption that was implemented by Groundspeak to make our lives harder.

**Reply by bekuno** [reply]
We have already a wiki site for the history: https://github.com/cgeo/cgeo/wiki/Project-roots-and-history

**Reply by rsudev** [reply]
https://github.com/just-radovan/c-geo

**Reply by Lineflyer** [reply]
> We have already a wiki site for the history: https://github.com/cgeo/cgeo/wiki/Project-roots-and-history

Oh yes, I forgot about that.

I would say:
Despite there was activity even before, if we want to have a "celebration date" it could be the first commit here (under correct license).
Maybe we could make some social media posting about that at least.

@rsudev I have the first commit located at July, 11th 2011 (see initial post). What makes you think its July, 10th 2010...or was that on the older repo (without correct license)?

**Comment by bekuno** [comment]
Adding the Apache License v2 was the last commit to the original repo https://github.com/just-radovan/c-geo/commit/d92c90355b43a0c12508888132bbeb3880dac89b on 6 Jul 2011

**Comment by fm-sys** [comment]
I think we could "celebrate" 10 years of c:geo in our changelog. This would IMHO be the right time currently:

"c:geo turns 10 years old. 🎉🎉🎉 

Many thanks to all who have supported c:geo within this time and have made c:geo available as free and powerful open source software. But we want to stay ready for the future. That's why we have integrated a new filter system into c:geo and have started redesigning the entire theming. We are far from finished, but we hope it will make c:geo even better. We see it as a kind of birthday present for c:geo ;-)"

What do you think?

**Reply by Lineflyer** [reply]
Maybe this would be the way in the changelog.
But for social media (and maybe homepage?) I would draft a small text with brief history where the app orginated and about the way until today focussing on the open source and community aspect and maybe even emphasize, that help is always welcome for all kind of activities (development, manual, graphics, testing, reporting bugs, etc. etc.)

**Comment by Lineflyer** [comment]
First early draft (e.g. for Facebook and other channels).
Feel free to correct, improve and fill in missing figures....feedback is welcome. This is just what came to my mind...needs fine tuning for sure.

---

**c:geo is celebrating its 10th birthday as an open source community project!**

The first c:geo app was originally published by carnero - the father of c:geo - back in 2010 as a hobby project. If you ever asked yourself why the app is called c:geo the reason is - as far as we know -, that it was a sort of abbreviation for carnero:geocaching.
After some time the effort for maintaining the app was just too much to handle it as a single person, so carnero planned to retire the project. 
But due to a lot of appreciation and offers to help from the geocaching community, carnero finally decided to publish the source code as open source and finally assigned the appropriate license in July 2011. An enthusiastic team of geocachers, with Java development skills, then started working on it. That was the starting point of c:geo as an open source project back in 2011.

Since then many people contributed to the project and helped c:geo to survive and continuously evolve and improve. 
After 10 years we are looking back to approximately:
- 15000 changes to the source code (commits) 
- 100 different contributors
- 500 active issues for bugs, enhancements and feature requests, and more than 6000 already closed ones
- 163 versions of the app

During the 10 years it has been an up and down in activity and contributors, which is typical for an open source project. Still we managed to keep c:geo alive  for the sake of the geocaching community throughout the time, which would have not been possible without our volunteers.
As of today c:geo consists of more than 270k lines of code and has a core team with 14 members (plus some additional contributors), providing support, development and keeping our servers up and running, which is just great!

We would like to take this opportunity to shout out a big THANK YOU!! to all contributors for their efforts and their help during the past 10 years!

At the same time we also would like to encourage the interested geocaching community to contribute to the project. We are always thankful for any kind of help, either in Java development or also in countless other areas like graphics, testing, support, web design, translations,  updating our user manual, and much more tasks.

**Reply by eddiemuc** [reply]
I like it!

**Reply by fm-sys** [reply]
>     * 11000 raised and 6000 closed issues for bugs or feature requests

This sounds like as 50% have not been solved yet. 11000 is the number issues + pullrequests. Furthermore I wouldn't call them closed but solved. 

But these are only minor things. Sounds already really nice in total 👍

**Reply by Lineflyer** [reply]
Any tool / page to find the total amount of different contributors of the repo?

**Reply by fm-sys** [reply]
Yes, see https://github.com/cgeo/cgeo/graphs/contributors

**Reply by Lineflyer** [reply]
Added contributors and LOC above.

**Reply by fm-sys** [reply]
> development team with 14 members (plus some additional contributors), which is just great!

In fact, there were exactly 99 contributors (+carnero?). So "some" is IMHO severely understated ;-) 

So I would write:

...development team with 14 members (and all in all [more than?] 100 different code contributors over the years), which is just great!

**Reply by Lineflyer** [reply]
That is already a few lines above ...100 different contributors. 
The sentence you refer to was meant to give the current situation.

**Reply by fm-sys** [reply]
Oh, yes. You're right, all fine 👍

Just the total issue number is still incorrect (as pull requests are no "issues" from user perspective & and the code contributions are already listed additionally). IMHO:

"Over 6000 solved issues and feature requests + 500 still waiting to be implemented or fixed"

**Reply by moving-bits** [reply]
Really like the text, thank you!
I've fixed some typos, clarified the issue counter and edited a the info about core team a bit.

**Reply by Narrat** [reply]
This sentence sounds kinda odd to me:
> After some time the effort for maintaining the app was just too much to handle it as a single person, so carnero planned to retire the project.

Reminds me of german :D
Maybe more like (although I don't know if this correct :D)
> After some time the effort of maintaining the app was just too much (to handle) for a single person, so carnero planned to retire the project.

Not sure about the part I put in braces, but imo it is also not necessary?

**Comment by Lineflyer** [comment]
As we missed the 10 years since the first commit to this repo,  I suggest I will post to social media on 16th August which is 10 years after the first open source released version from this repo.

I will put up a scheduled posting on Facebook in order not to forget it.
Anyone out there willing and able to design a anniversary logo or graphic...like a c:geo icon with birthday hat or alike?

**Reply by fm-sys** [reply]
I could do so. Maybe something like that (just a quick draft):

![Screenshot_20210724_234004](https://user-images.githubusercontent.com/64581222/126881604-83119dfc-7470-46e1-b561-1ffe90300b93.jpg)

I still don't like it myself. Probably we need to either leave out the face or the logo foreground...

**Reply by eddiemuc** [reply]
I like it :-)

**Reply by Lineflyer** [reply]
Maybe just our logo with some sparkles around it and a 10 somewhere?

**Comment by Lineflyer** [comment]
I scheduled a posting for 16.08.2021 11:00 UTC+02 using the logo and the text above on Facebook.

**Reply by fm-sys** [reply]
Sorry, that I didn't manage it to create a good graphic earlier. But here we are:

@Lineflyer: Please give me a short notice if you have read this reply
@all: Please tell me if you like this graphic (inspirited by our current functional graphic, which I had designed some time ago)


![10th birthday](https://user-images.githubusercontent.com/64581222/128931660-e02b7d41-b2b1-4378-92c7-3302c7dcf13e.png)

**Reply by fm-sys** [reply]
# V2

...just have moved the emojis and description text a bit around...

![10th birthday - v2](https://user-images.githubusercontent.com/64581222/128932826-a11f05a8-fbae-4d16-9420-6da42a70e799.png)

**Reply by Lineflyer** [reply]
@fm-sys  Read it! I changed the scheduled posting to a normal posting (removed the image and used a facebook custom smiley background)  and I will place your proposal as a temporary title picture on Feacbook on the same date for a few days/weeks

**Comment by fm-sys** [comment]
Another idea: Don't know if it's possible without much effort, what but what if we add the graphic as starting image to our changelog for the "big UI release"?

**Comment by Lineflyer** [comment]
All done. Thanks for wording and graphics!!
Many positive reactions on social media.
Not sure whether its readable without login: https://www.facebook.com/android.geocaching/posts/10159048680285091

**Reply by moving-bits** [reply]
seems to be not readable without account/login :-(

**Reply by Lineflyer** [reply]
Small screenshot.
![image](https://user-images.githubusercontent.com/949669/129625526-c9232dd5-5758-4c23-88a2-a4aa7703b756.png)

**Reply by moving-bits** [reply]
Nice messages! Thanks for providing this insight.

**Comment by bekuno** [comment]
> The first tagged release was for 26th, August 2011 with  `market_20110826`  https://github.com/cgeo/cgeo/releases/tag/market_20110826

I took a look at the history today.
The first release was https://github.com/cgeo/cgeo/commit/ee4e9ae6368f1baaaa879d7ddb31a32d25208223 on 2011-08-15, the second was https://github.com/cgeo/cgeo/commit/a0b181b09647304ca86b984c805c4fefa5b19099 on 2011-08-19. We have no tag for both.

With the new release 2021-08-15 we hit the 10th birthday perfectly.
